pyirf/io/gadf.py

```diff
@@ -325,3 +309,177 @@ def create_rad_max_hdu(
+        bins
+    """
+
+    tag_lo = tag + '_LO'
+    tag_hi = tag + '_HI'
+
+    # to allow the function work on either single file or a list of files convert a single string into a list
+    if isinstance(file_name, str):
+        file_name = [file_name]
+
+    old_table = None
+    for this_file in file_name:
+        table = QTable.read(this_file, hdu=extname)
+        if old_table is not None:
+            if not old_table[tag_lo][0].shape == table[tag_lo][0].shape:
+                raise ValueError('Non matching bins')
+            if not ((old_table[tag_lo][0] == table[tag_lo][0]).all() and (old_table[tag_hi][0] == table[tag_hi][0]).all()):
+                raise ValueError('Non matching bins')
+        old_table = table
+
+    return table[tag_lo][0], table[tag_hi][0]
+
+
+def read_irf_grid(files, extname, field_name):
+    """
+    Reads in a grid of IRFs for a bunch of different parameters and stores them in lists
+
+    Parameters
+    ----------
+    files: string or list of strings
+        files to be read
+    extname: string
+        name of the extension to read the data from in fits file
+    field_name: string
+        name of the field in the extension to extract
+
+    Returns
+    -------
+    irfs_all: np.array
+        array of IRFs, first axis specify the file number(if more then one is given),
+        second axis is the offset angle, rest of the axes are IRF-specific
+    theta_bins: astropy.units.Quantity[angle]
+        theta bins
+    """
+
+    # to allow the function work on either single file or a list of files convert a single string into a list
+    if isinstance(files, str):
+        files = [files]
+
+    n_files = len(files)
+
+    irfs_all = np.empty(n_files, dtype=np.object)
+    for ifile, this_file in enumerate(files):
+        # [0] because there the IRFs are written as a single row of the table
+        irfs_all[ifile] = QTable.read(this_file, hdu=extname)[field_name][0]
+
+    # if the function is run on single file do not need the first axis dimension
+    if n_files == 1:
+        irfs_all = irfs_all[0, ...]
+    # the last operation converts an array of objects to a multidimentional table
+    return np.array(irfs_all.tolist())
+
+
+def read_aeff2d_hdu(file_name, extname="EFFECTIVE AREA"):
+    """
+    Reads effective area from FITS file
+
+    Parameters
+    ----------
+    file_name: string or list of strings
+        file(s) to be read
+    extname:
+        Name for BinTableHDU
+
+    Returns
+    -------
+    effective_area: astropy.units.Quantity[area]
+        Effective area array of shape (n_energy_bins, n_fov_offset_bins) or (n_files, n_energy_bins, n_fov_offset_bins)
+    true_energy_bins: astropy.units.Quantity[energy]
+        Bin edges in true energy
+    fov_offset_bins: astropy.units.Quantity[angle]
+        Bin edges in the field of view offset.
+    """
+
+    field_name = "EFFAREA"
+    effective_area = read_irf_grid(file_name, extname, field_name)
```

Comment:
please check my note at the top of the issue, this is done such that you can also load a bunch of IRFs in one go
working on such lists of IRFs will be needed in many points of the code, and it is not as trivial because you have a multidimentional objects here so it is good to have it done in one generic function then to deal with it every time you load some kind of IRFs

Reply:
Yes, and I don't think it's a good idea.

Reply:
well, the functionality is needed, because we will need to read a bunch of IRFs for interpolation/averaging. If we do not implement functions to read lists of IRFs they such reading would appear in many places of the code resulting in code repetition. 
But if you have some concrete suggestion how to rewrite it keeping the same functionality and avoiding code repetition for each type of IRF please let me know and I can implement this